buildSrc/src/main/java/org/opensearch/gradle/docker/DockerSupportService.java

```diff
@@ -68,17 +69,30 @@ public abstract class DockerSupportService implements BuildService<DockerSupport
     // Defines the possible locations of the Docker CLI. These will be searched in order.
     private static String[] DOCKER_BINARIES_UNIX = { "/usr/bin/docker", "/usr/local/bin/docker" };
 
+    // On Mac you can install docker and docker-compose via brew adding more options where the binaries can be located.
+    private static String[] DOCKER_BINARIES_MACOS = Stream.concat(
```

Comment:
`/opt` isn't mac specific. Not sure we need the `DOCKER_BINARIES_MACOS`. Maybe make this `DOCKER_BINARIES_HOMEBREW` and add `/opt/homebrew/bin/docker`, and `/home/linuxbrew/.linuxbrew` per the [homebrew documentation](https://docs.brew.sh/Installation)?

Reply:
@nknize I think this is fair point. Let me update the PR.

Reply:
@nknize Thinking more about this, depending on how robust we want to make it but why don't we simply use `which docker` and `which docker-compose` on Linux and MacOS based systems instead? My gut tells me there can be always some edge case where docker (or other binary in general) can be installed in custom location...

What is the chance that `which docker` fails (i.e. gives no useful answer) if the `docker` is installed on the system?

If I understand the code correctly then the only change/implication would be that the paths to search in would not be static class members but they will have to be evaluated only after the `ExecOperations execOperations` is injected into the SupportService ctor. That shouldn't be a big deal... WDYT?

Reply:
I like that idea. How about using the static variables as a fall back if `which docker` does fail? And since OS X is unix based anyway, you should just be able to add the new homebrew paths to the existing `DOCKER_BINARIES_UNIX` variable instead of a new `DOCKER_BINARIES_MACOS`?

Reply:
Hi @nknize,

I updated my PR and I think it is ready for review.

I decided to not include the `which` functionality into this PR for now because it turns out to be more complicated. Basically the `which` output can differ based on host systems quite a lot (these are more or less corner cases but we need to deal with them as well, things like "aliases" or "shell functions"). This would lead to more complex parsing of the host output than I like for this PR. I will open a new ticket to enable discussion about it and provide some of the findings I made. In the end I think we will need to add functionality where users will define their own path to container engine anyway (think about people who have both the `docker` and `podman` installed at the same time [yes, I do believe we will have `podman` build compatibility in the future 😄 ]).

For now, let this PR address only the Homebrew locations.